crates/hyperqueue/src/server/job.rs

```diff
@@ -207,6 +212,10 @@ impl Job {
         self.tasks.len() as JobTaskCount
     }
 
+    pub fn has_no_active_tasks(&self) -> bool {
+        self.counters.is_terminated(self.n_tasks())
+    }
+
     pub fn is_terminated(&self) -> bool {
         !self.is_open && self.counters.is_terminated(self.n_tasks())
```

Comment:
We can use `has_no_active_tasks` here, just to make sure it does not get desynchronized.